src/server/service/user.ts

```diff
@@ -30,7 +30,51 @@ export default class UserService implements IUserService {
     return hash;
   }
 
-  async create(dto: CreateUserDTO): Promise<User> {
+  private checkRoles(role: string): void {
+    const ROLES = [
+      Role.Admin as string,
+      Role.Manager as string,
+      Role.User as string,
+    ];
+
+    if (!ROLES.includes(role)) {
+      throw new InvalidUserRole();
+    }
+  }
+
+  private async checkPermission(email: string, role: Role) {
```

Comment:
Why receiving the `email` if we already fetched the whole user in L78? I think we could receive just the role of the creator here and the role to create.

```suggestion
  private async checkPermission(issuerRole: Role, expectedRole: Role) {
```

Reply:
Another observation is that this operation is only performed when creating a user, thus it doesn't make much sense to have it isolated.

You want to check for higher roles to create lower ones only when creating a user, why moving it out of there?